### PR TITLE
Add test for fb with mismatched attachment rb/tex-targets.

### PR DESCRIPTION
**Pull request by jdashg**
Addresses #2558.

### CODE REVIEW

**Comment by kenrussell** [comment]
To be clear: the above failures are on macOS 10.15.6, using the system's OpenGL implementation. ANGLE isn't involved in Chrome's stack, though I tested both with and without it.

**Comment by jdashg** [comment]
(Addresses #2558, since having it in the title didn't auto-link it)

**Comment by kenrussell** [comment]
Good point @jdashg - updated the title and description so they'll look good in the eventual commit message.

Are you planning to push an update which allows FRAMEBUFFER_UNSUPPORTED for some or all of these combinations? If so, then we could go ahead and merge this, since https://bugs.chromium.org/p/angleproject/issues/detail?id=5009 tracks fixing the test on ANGLE on Windows.

**Comment by jdashg** [comment]
PTAL

File: sdk/tests/conformance2/rendering/framebuffer-mismatched-attachment-targets.html [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  const fb = gl.createFramebuffer();
+  gl.bindFramebuffer(gl.FRAMEBUFFER, fb);
+  for (const color of colorResList) {
+    for (const depth of depthResList) {
+      debug(`\ncolor: ${color.target}; depth: ${depth.target}`);
+      for (let mipLevel = 0; mipLevel < MIP_LEVELS; mipLevel++) {
+        debug(`mipLevel: ${mipLevel}`);
+        color.attach(gl.COLOR_ATTACHMENT0, mipLevel);
+        depth.attach(gl.DEPTH_ATTACHMENT, mipLevel);
+        if (!wtu.framebufferStatusShouldBe(gl, gl.FRAMEBUFFER, [gl.FRAMEBUFFER_COMPLETE, gl.FRAMEBUFFER_UNSUPPORTED])) {
```
Comment (kenrussell) [inline sdk/tests/conformance2/rendering/framebuffer-mismatched-attachment-targets.html:135]: The cases mentioned earlier still fail on macOS, including in Firefox Nightly. This has to continue here (skip the rendering and clearing tests) if the framebuffer status is FRAMEBUFFER_UNSUPPORTED.

Comment (jdashg) [reply sdk/tests/conformance2/rendering/framebuffer-mismatched-attachment-targets.html:135]: It's sort of a pain to spin up a machine to fix this. If you see what's wrong it's most efficient if you can suggest/provide a fix collaboratively!

Comment (kenrussell) [reply sdk/tests/conformance2/rendering/framebuffer-mismatched-attachment-targets.html:135]: Got it. I'll post a diff here shortly which you can apply.

Comment (kenrussell) [reply sdk/tests/conformance2/rendering/framebuffer-mismatched-attachment-targets.html]: Never mind, your revision is more elegant than the one I was going to propose. :)

File: sdk/tests/conformance2/rendering/framebuffer-mismatched-attachment-targets.html [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+</head>
+<body>
+<canvas id="example" width="1", height="1"></canvas>
+<div id="description"></div>
+<div id="console"></div>
+<script>
+"use strict";
+debug("");
+
+description("Test that WebGL2 can render to layers in 3D textures");
```
Comment (jdarpinian) [inline sdk/tests/conformance2/rendering/framebuffer-mismatched-attachment-targets.html:37]: Description needs updating.

**Review by jdarpinian — APPROVED** [review]

**Comment by kenrussell** [comment]
@jdashg please merge yourself once @jdarpinian 's review feedback is addressed. Thanks!

**Pull request merged**